addons/event/models/event_event.py

```diff
@@ -282,47 +298,70 @@ def _compute_start_sale_date(self):
+                event.date_tz = self.env.user.tz or 'UTC'
+
+    @api.depends('event_type_id')
+    def _compute_seats_availability(self):
+        """ Make it separate from ``_compute_from_event_type`` because otherwise
+        a value given at create (see create override) would protect all other fields
+        depending on event type id from being computed (see ``_field_computed``) """
+        for event in self:
+            if event.event_type_id.default_registration_max:
+                event.seats_availability = 'limited'
+            if not event.seats_availability and not event._origin.seats_availability:
+                event.seats_availability = 'unlimited'
+
+    @api.depends('event_type_id')
+    def _compute_from_event_type(self):
+        for event in self:
+            if not event.event_type_id:
+                if not event.seats_availability:
+                    event.seats_availability = 'unlimited'
+                if not event.is_online:
+                    event.is_online = False
+                continue
 
-            if self.event_type_id.auto_confirm:
-                self.auto_confirm = self.event_type_id.auto_confirm
+            if event.event_type_id.default_registration_max:
+                event.seats_max = event.event_type_id.default_registration_max
 
-            if self.event_type_id.use_timezone:
-                self.date_tz = self.event_type_id.default_timezone
+            if event.event_type_id.auto_confirm:
+                event.auto_confirm = event.event_type_id.auto_confirm
 
-            self.is_online = self.event_type_id.is_online
+            event.is_online = event.event_type_id.is_online
 
-            if self.event_type_id.use_mail_schedule and self.event_type_id.event_type_mail_ids:
-                self.event_mail_ids = [(5, 0, 0)] + [
+            # compute mailing information
+            if event.event_type_id.use_mail_schedule and event.event_type_id.event_type_mail_ids:
+                event.event_mail_ids = [(5, 0, 0)] + [
                     (0, 0, {
                         attribute_name: line[attribute_name] if not isinstance(line[attribute_name], models.BaseModel) else line[attribute_name].id
                         for attribute_name in self.env['event.type.mail']._get_event_mail_fields_whitelist()
                         })
-                    for line in self.event_type_id.event_type_mail_ids]
+                    for line in event.event_type_id.event_type_mail_ids]
 
             # compute tickets information
-            if self.event_type_id.use_ticket:
-                all_ticket_values = []
-                for ticket in self.event_type_id.event_type_ticket_ids:
-                    ticket_vals = dict(
-                        (attribute_name, ticket[attribute_name] if not isinstance(ticket[attribute_name], models.BaseModel) else ticket[attribute_name].id)
+            if event.event_type_id.use_ticket:
```

Comment:
We don't check for `event.event_type_id.event_type_ticket_ids` here.

Seems a bit incoherent with `event.event_type_id.event_type_mail_ids` just above.

Reply:
Indeed. I updated code. Note that this is a small functional change compared to pre-compute: previously we did not reset mailing information on event type when unchecking use_mailing. Now it is done. Probably worth a note somewhere.